next/src/pages/settings.tsx

```diff
@@ -13,11 +35,77 @@ const SettingsPage = () => {
         label="Languages"
         value={language}
         valueMapper={(e) => e.name}
-        onChange={setLanguage}
+        onChange={(e) => updateSettings(SETTINGS_LANGUAGE, e)}
         items={languages}
       />
+      <Combo<GPTModelNames>
+        label="Models"
+        value={modelName}
+        valueMapper={(e) => e}
+        onChange={(e) => updateSettings(SETTINGS_MODEL_NAME, e)}
+        items={GPT_MODEL_NAMES}
+      />
+
+      <h1 className="mt-6">Advanced Settings</h1>
+      <Input
+        label={`${t("TEMPERATURE", { ns: "settings" })}`}
+        value={temperature}
+        name="temperature"
+        type="range"
+        onChange={(e) => updateSettings(SETTINGS_TEMPERATURE, parseFloat(e.target.value))}
+        attributes={{
+          min: 0,
+          max: 1,
+          step: 0.01
+        }}
+        helpText={`${t("HIGHER_VALUES_MAKE_OUTPUT_MORE_RANDOM", {
+          ns: "settings",
+        })}`}
+      />
+      <Input
+        label={`${t("LOOP", { ns: "settings" })}`}
+        value={maxLoops}
+        name="loop"
+        type="range"
+        onChange={(e) => updateSettings(SETTINGS_MAX_LOOPS, parseFloat(e.target.value))}
+        attributes={{
+          min: 1,
+          max: 25,
+          step: 1
+        }}
+        helpText={`${t("CONTROL_THE_MAXIMUM_NUM_OF_LOOPS", {
+          ns: "settings",
+        })}`}
+      />
+      <Input
+        label={`${t("TOKENS", { ns: "settings" })}`}
+        value={maxTokens}
+        name="tokens"
+        type="range"
+        onChange={(e) => updateSettings(SETTINGS_MAX_TOKEN, parseFloat(e.target.value))}
+        attributes={{
+          min: 200,
+          max: 2000,
+          step: 100
+        }}
+        helpText={`${t("CONTROL_MAXIMUM_OF_TOKENS_DESCRIPTION", {
+          ns: "settings",
+        })}`}
```

Comment:
Since this already returns a string, we don't need to wrap this in a paramaterized string:
```suggestion
        helpText={t("CONTROL_MAXIMUM_OF_TOKENS_DESCRIPTION", { ns: "settings"})}
```